src/corehost/src/main.cpp

```diff
@@ -179,46 +203,26 @@ int main(const int argc, const pal::char_t* argv[])
     trace::info(_X("App argc: %d"), args.app_argc);
     trace::info(_X("App argv: %s"), argstr.c_str());
 
-    auto app_base = get_directory(args.managed_application);
-    auto app_name = get_filename(args.managed_application);
-
-    // App-local coreclr wins
-    {
-        pal::string_t candidate;
-        candidate.assign(app_base);
-        append_path(candidate, LIBCORECLR_NAME);
-
-        if (pal::file_exists(candidate))
-        {
-            args.clr_path.assign(app_base);
-        }
-    }
-
-    if (args.clr_path.empty())
+    pal::string_t clr_path;
+    if (!resolve_clr_path(args, &clr_path))
     {
-        trace::error(_X("failed to locate CLR files, set the DOTNET_HOME environment variable"));
+        trace::error(_X("could not resolve coreclr path"));
         return 1;
```

Comment:
Can I add this as part of another PR? Issue: https://github.com/dotnet/cli/issues/627